src/main/kotlin/net/yested/core/properties/properties.kt

```diff
@@ -13,24 +14,39 @@ private interface PropertyChangeListener<in T> {
 interface ReadOnlyProperty<out T> {
     fun get():T
     fun onNext(handler: (T)->Unit):Disposable
+    fun detectContentChange()
 }
 
 class Property<T>(initialValue: T): ReadOnlyProperty<T> {
 
     private var value : T = initialValue
-    private var valueHashCode = value?.hashCode()
+    private var valueHash = getValueHash(value)
     private val listeners = mutableSetOf<PropertyChangeListener<T>>()
 
     fun set(newValue: T) {
-        // check the hashCode in case the value is mutable, the same instance is provided, but its contents have changed.
-        val newValueHashCode = newValue?.hashCode()
-        if (newValue != value || newValueHashCode != valueHashCode) {
+        // check the value hash in case the value is mutable, the same instance is provided, but its contents have changed.
+        val newValueHash = getValueHash(newValue)
+        if (newValue != value || newValueHash != valueHash) {
```

Comment:
Can we change this into:
if (newValue != null || getValueHash(newValue) != getValueHash(value)) { ... }
I think that in most cases users will put into another instance, hence reference comparison would work in 90% cases. At least in my app I put in always difference instances. Even if it is mutated collection.